crates/sui-core/src/gateway_state.rs

```diff
@@ -670,20 +674,30 @@ where
     }
 
     async fn retry_pending_tx(&self, digest: TransactionDigest) -> Result<(), anyhow::Error> {
-        let tx = self
-            .store
-            .get_transaction(&digest)?
-            .ok_or(SuiError::TransactionNotFound { digest })?;
-        self.execute_transaction(tx).await?;
-        Ok(())
+        let tx = self.store.get_transaction(&digest)?;
+        match tx {
+            Some(tx) => {
+                self.execute_transaction(tx).await?;
+                Ok(())
+            }
+            None => {
+                // It's possible that the tx has been executed already.
+                if self.store.get_certified_transaction(&digest)?.is_some() {
```

Comment:
This section should have been written with `get_certified_transaction(&digest).map_err(...)?`